test/albumController.test.js

```diff
@@ -0,0 +1,92 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Album = require('../src/models/Album');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('AlbumController API Tests', () => {
+  const mockAlbumData = {
+    albumName: 'Test Album',
+    artistName: 'Test Artist',
+    price: 9.99,
+    spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+  };
+
+  beforeEach(async () => {
+    await Album.deleteMany({});
+    await Album.create(mockAlbumData);
+  });
+
+  it('should test the getAllAlbums endpoint - Success Case', async () => {
+    const response = await request(app).get('/api/v1/albums');
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body.albums).toHaveLength(1);
+  });
+
+  it('should return an empty list if there are no albums in the database', async () => {
+    await Album.deleteMany({});
+    const response = await request(app).get('/api/v1/albums');
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toEqual([]);
```

Comment:
The error message given by the test shows that the `body` of the response isn't directly an empty array; verify this against the swagger docs/controller code as well to see what we should be asserting here:

```
  ● AlbumController API Tests › should return an empty list if there are no albums in the database

    expect(received).toEqual(expected) // deep equality

    Expected: []
    Received: {"albums": [], "count": 0}

      47 |     const response = await request(app).get('/api/v1/albums');
      48 |     expect(response.status).toBe(StatusCodes.OK);
    > 49 |     expect(response.body).toEqual([]);
         |                           ^
      50 |   });
      51 |
      52 |   it('should test the getSingleAlbum endpoint - Success Case', async () => {

      at Object.toEqual (test/albumController.test.js:49:27)
```

Reply:
I tried to edit it in this way commit d565fd8fd84ddca98a0a6da9e84803aebe2cbf1e